### PR TITLE
added support for M0

### PR DESCRIPTION
**Pull request by Adelmar92**
## Proposed changes

added support for for M0:

- Added new Provider Name "Tenant"
- Added new proerty required for Tenant, "BackingProviderName"
- Added validation to avoid calling the ProcessArgs when AdvancedSettings is not present
- Added example file

_Jira ticket:_ [INTMDB-905](https://jira.mongodb.org/browse/INTMDB-905)

## Testing:

cfn testing: 
image

image

image

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [x] cfn invoke for each of CRUDL/cfn test
- [x] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [x] Published to AWS private registry
- [x] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [x] Deleted stack to ensure resources are deleted
- [x] Created multiple resources in same stack
- [x] Validated in Atlas UI
- [x] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

File: cfn-resources/cluster/cmd/resource/resource.go [hunk 1]
```diff
@@ -369,20 +365,45 @@ func clusterCallback(client *mongodbatlas.Client, currentModel *Model, projectID
...
 		_, _ = log.Debugf("Updating cluster settings:%s", *currentModel.Name)
 		return updateClusterSettings(currentModel, client, projectID, cluster, &progressEvent)
 	}
 	return progressEvent, nil
 }
 
+func (m *Model) HasAdvanceSettings() bool {
+	/*TODO: this logic is because of a bug un Cloud Formation, when we return in_progress in the CREATE
+	,the second time the CREATE gets executed
+	it returns the AdvancedSettings is not nil but its fields are nil*/
```
Comment (andreaangiolillo) [inline cfn-resources/cluster/cmd/resource/resource.go:394]: Can you create an INTMDB ticket and explain what to do to remove this TODO?

Comment (Adelmar92) [reply cfn-resources/cluster/cmd/resource/resource.go:394]: is a super weird cfn behaviour

File: cfn-resources/cluster/cmd/resource/resource.go [hunk 2]
```diff
@@ -389,9 +389,6 @@ func clusterCallback(client *mongodbatlas.Client, currentModel *Model, projectID
 }
 
 func (m *Model) HasAdvanceSettings() bool {
-	/*TODO: this logic is because of a bug un Cloud Formation, when we return in_progress in the CREATE
-	,the second time the CREATE gets executed
-	it returns the AdvancedSettings is not nil but its fields are nil*/
```
Comment (andreaangiolillo) [inline cfn-resources/cluster/cmd/resource/resource.go]: I think we should keep the message to explain why there is this logic. Just remove the `TODO` from it

Comment (Adelmar92) [reply cfn-resources/cluster/cmd/resource/resource.go]: agreed! ill just add a ticket so we can keep track of this issue, ill try to create one on AWS if i dont het an answer on chime

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**